src/Data/List/Relation/Suffix/Heterogeneous/Properties.agda

```diff
@@ -0,0 +1,233 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of the heterogeneous suffix relation
+------------------------------------------------------------------------
+
+module Data.List.Relation.Suffix.Heterogeneous.Properties where
+
+open import Function using (_$_; flip)
+open import Relation.Nullary using (Dec; yes; no)
+import Relation.Nullary.Decidable as Dec
+open import Relation.Unary as U using (Pred)
+open import Relation.Nullary.Negation using (contradiction)
+open import Relation.Binary as B using (REL; Rel; Trans; Antisym; Irrelevant; _⇒_)
+open import Relation.Binary.PropositionalEquality as P using (_≡_; refl; sym; subst)
+open import Data.Nat as N using (suc; _+_; _≤_; _<_)
+open import Data.List as List using (List; []; _∷_; _++_; length; filter; replicate; reverse)
+open import Data.List.Relation.Pointwise as Pw using (Pointwise; []; _∷_; Pointwise-length)
+open import Data.List.Relation.Suffix.Heterogeneous as Suffix using (Suffix; here; there; tail)
+open import Data.List.Relation.Prefix.Heterogeneous as Prefix using (Prefix)
+import Data.Nat.Properties as ℕₚ
+import Data.List.Properties as Listₚ
+import Data.List.Relation.Prefix.Heterogeneous.Properties as Prefixₚ
+
+------------------------------------------------------------------------
+-- reverse (convert to and from Prefix)
+
+module _ {a b r} {A : Set a} {B : Set b} {R : REL A B r} where
+
+  -- TODO: move this lemma (and others) into Pointwise?
```

Comment:
Yes! It would be nice if these were expressed in terms of `reverseAcc`
which I am introducing in #529.

So `pw-reverse` would be `reverse⁺` and `foldl` would be `reverseAcc⁺`.